docs/next_link_analysis.md

```diff
@@ -0,0 +1,108 @@
+
+### Conditioning on time
+
+As we are focussing on the single hour of data, we have implicitly conditioned on time. Different  predictions would be made for different slices of time. This would mean that this approach could incoporate non-stationary effects and the impact of current events. As will be discussed beneath, the model could be run in real time, as it is effectively relying on aggregations and dictionary data access which is inherently fast.
+
+### Conditioning on location
+
+Location is not considered here, but would be possible given that there is information about geolocation of the user. This simply requires consideration of aggregation weights as a function of distance, discounting information from users further away.
+
+## Preparing the lagged data
+
+The data requires a lagged URL to be created in order to see the link between current and past data. We are trying to predict the URL to be visited, by referencing the lagged URL.
+
+Of note, a full 53% of the users on the government website only visit one link. This makes it impossible to create a lagged URL. These users data are removed from the dataset, as we cannot build a prediction conditioned on the past, as there is no history for the user.
+
+### Removing login, search and done activity
+
+Any URLs for search or login are removed. Search activity only accounts for [4%] of the remaining data, which when considered in combination with the number of users who only visit one URL, is indicative of external search engines.
+
+Login activity, [6%] of the data, was removed as its a deterministic intermediate link to certain URLs, ie the probability of `P(a->login->b)` is the same as `P(a->b)`, as `P(login->b)` is 100%. Which in turn would mean that it obscures the relationship `P(a -> b)`. This admittedly ignores the losses from people who forget their login details.
+
+Done activity was also removed, approximately a further [1%] of the data. The rationale for removing done activity was that it just represents the completion of an activity or form and is not helping the user access information.
+
+### Circular journeys
+
+Where the next URL is the same as the preceding URL, the data is removed.
+
+### Aggregated data - top sites
+
+![top sites by count](./images/top_sites.png)
+
+### Aggregated data - some low count sites
+
+![example low count sites](./images/bottom_sites.png)
+
+As seen, some sites have only one single visit in the data. There are 34k unique URLs in the data, in the end.
+
+## Model build
+
+### Train test split
+
+In order to build the model, the data is split using stratified sampling, so that 70% of users are in the train data, and the remainder in the test. No consideration was made for the actual time. This means that higher frequency effects than a one hour period are being ignored.
```

Comment:
What are we stratifying by? Taxons?

For instance, if most of the train set is on HMRC pages, then the model is more likely to predict a HMRC related page as the next page.